### ISSUE TITLE
Use a Node test framework for unit tests

### ISSUE THREAD

**Issue by Jameskmonger** [courtroom/courtroom#41]
I think that testing with Karma is unnecessary at the moment, we should instead be using a Node testing framework as we don't need to run a browser testing framework.

I think Mocha would be a good fit, but I'm not sure about its compatibility with Jasmine.

**Comment by Jameskmonger** [comment]
@SiAdcock I've been attempting to use `mocha` with `expect.js` in the v0.8 branch, but it doesn't seem to be running any tests on the Travis server. Any chance you could take a look?

**Comment by SiAdcock** [comment]
Definitely agree with ditching karma for something more node-y. In browser tests are not necessary for this library IMO.

Mocha is a better choice than Jasmine because Jasmine is an all-or-nothing framework / runner / assertion library. Jasmine just does too many things.

Mocha is just a framework. You can bring your own assertions (like expect.js, you've already discovered. Chai, another good one). Do one thing and do it well should be a Node developer's motto. 

The issue looks like it's in the Gulpfile. You're building the test specs, then building the source then running the tests. But all of this happens asynchronously, so `test-build/spec/**/*.spec.js` probabaly don't exist when you pipe them through Mocha.

Have a look at the old Gulpfile for an example of how to work around this:

```javascript
gulp.task('test', ['test-build:spec:commonjs', 'test-build:src:commonjs'], () => {
    return gulp.src('test-build/spec/**/*.spec.js', {read: false})
        .pipe(mocha());
});

gulp.task('test-build:spec:commonjs', (done) => {
  const tsProjectCommonJS = tsc.createProject('spec/tsconfig.json', {
    module: 'commonjs'
  });

  tsProjectCommonJS.src()
    .pipe(tsc(tsProjectCommonJS))
    .pipe(gulp.dest('test-build'))
    .on('end', done);
});

gulp.task('test-build:src:commonjs', (done) => {
  const tsProjectCommonJS = tsc.createProject('src/tsconfig.json', {
    module: 'commonjs'
  });

  tsProjectCommonJS.src()
    .pipe(tsc(tsProjectCommonJS))
    .pipe(gulp.dest('test-build/src'))
    .on('end', done);
});
```

Pass a `done` callback through to the `test-build` tasks. The callback is executed when the `'end'` event is triggered. When you call the `test` task, it performs the two `test-build` tasks and only when they are complete does it executed the final callback that pipes the specs into Mocha.

Let me know if this doesn't make sense, or if it doesn't solve the issue!

**Comment by Jameskmonger** [comment]
Fixed by #42. Thanks so much, @SiAdcock !
